### PR TITLE
util/zlib/deflater: Fix several compiler warnings.

### PR DESCRIPTION
**Pull request by besser82**
As the title says, just some small fixes.

### CODE REVIEW

File: util/zlib/deflater.c [hunk 1]
```diff
@@ -84,7 +84,8 @@ int main(int argc, char* argv[])
         }
 
         /* display summary */
-        printf("Compressed %s (%d bytes) to %s (%d bytes)\n",
-                argv[1], inlen, argv[2], outlen);
+        printf("Compressed %s (%lu bytes) to %s (%lu bytes)\n",
+                argv[1], (unsigned long) inlen,
+                argv[2], (unsigned long) outlen);
```
Comment (greg-king5) [inline util/zlib/deflater.c]: cc65's `printf()` family of functions know the `h, j, t,` and `z` size modifiers.  Those functions don't know the `hh` modifier.

Comment (besser82) [reply util/zlib/deflater.c]: Fixed in rebased commit.

**Comment by oliverschmidt** [comment]
Sorry but could one of you both elaborate a bit?

- According to my understanding chars are by default unsigned with cc65.
- https://github.com/cc65/cc65/blob/master/include/zlib.h is about chars, not unsigned chars.

**Comment by besser82** [comment]
From my understanding this program is considered to be compiled and run on a system, that has a 'real' zlib available to compress data, that can later be decompressed using cc65's zlib `inflatemem()`.  Those systems are not guaranteed to have `unsigned char` as default, nor `size_t` to be defined as an `unsigned int`.

**Comment by groessler** [comment]
It's never bad to explicitly put the signedness of a char variable in its definition ("signed char" or "unsigned char").
Don't know if the mixed chars in `zlib.h` are on purpose or accidently there.

**Comment by oliverschmidt** [comment]
>From my understanding this program is considered to be compiled and run on a system, that has a 'real' zlib available to compress data, that can later be decompressed using cc65's zlib inflatemem().

Thanks for the clarification.

> Those systems are not guaranteed to have unsigned char as default, nor size_t to be defined as an unsigned int.

This isn't guaranteed with cc65 either. There's the `--signed-char` cmdline option. From that perspective it becomes clear why there are so many `unsigned char` declaration in cc65 header files.

So if you think the variables in question have to be unsigned then the same holds for the function declarations in https://github.com/cc65/cc65/blob/master/include/zlib.h, the code in https://github.com/cc65/cc65/blob/master/libsrc/zlib/uncompress.c (and the comment in https://github.com/cc65/cc65/blob/master/libsrc/zlib/inflatemem.s).

Can you imagine to adjust those locations too?

**Comment by besser82** [comment]
> Can you imagine to adjust those locations too?

Done in https://github.com/cc65/cc65/pull/913/commits/364b3a575b0a391918b918880d875e13c4ae72b3.

**Comment by oliverschmidt** [comment]
Thanks for taking care of my request and for your patience. Merging now...

**Pull request merged**